enterprise/server/backends/authdb/authdb.go

```diff
@@ -25,19 +29,106 @@ import (
 
 const (
 	apiKeyLength = 20
+
+	// For encrypted keys, the first N characters serve as the nonce.
+	// This cannot be changed without a migration.
+	apiKeyNonceLength = 6
+
+	apiKeyEncryptionBackfillBatchSize = 100
 )
 
 var (
 	userOwnedKeysEnabled = flag.Bool("app.user_owned_keys_enabled", false, "If true, enable user-owned API keys.")
+	apiKeyEncryptionKey  = flag.String("auth.api_key_encryption.key", "", "Base64-encoded 256-bit encryption key for API keys.")
+	encryptNewKeys       = flag.Bool("auth.api_key_encryption.encrypt_new_keys", false, "If enabled, all new API keys will be written in an encrypted format.")
+	encryptOldKeys       = flag.Bool("auth.api_key_encryption.encrypt_old_keys", false, "If enabled, all existing unencrypted keys will be encrypted on startup.")
```

Comment:
nit: should we add a note here that the old `value` column has to be manually cleared out once this is done?

Reply:
done